### DISCUSSION TITLE
Multiple IP Addresses per Profile

### DISCUSSION THREAD

**Discussion by KurtisLiggett** [snappysnappydog/Simple-IP-Config#110] [Ideas]
Allow option to add multiple IP addresses to each profile.
-Similar to 'Advanced...' button in Windows' built-in configuration screen.

**Comment by giacomocerquone** [comment]
Sorry but I can't understand this... multiple ip address per profile shouldn't/couldn't be stored! What do you really mean?

**Comment by KurtisLiggett** [comment]
**I really mean multiple addresses per profile!**

Here is my vision for the GUI:
The GUI would be switched to a new mode showing listviews for the IP/DNS addresses. The columns would be edited by clicking on the address you want to edit in-place. It would have a add/delete buttons to add addresses.
I will post a drawing once I have time.

The ini file will store addresses like this
IpAddress=192.168.1.1|10.225.251.12
IpSubnet=255.255.255.0|255.255.255.0

It would be similar to the built-in Windows functionality, but easier to use.
Something like this:
![image](https://user-images.githubusercontent.com/29669372/28476515-e02f1b68-6e15-11e7-99ee-5b9cc1298b7c.png)


*I haven't marked this as 'Accepted' because it is a lot of work and I don't personally need this feature.

**Comment by giacomocerquone** [comment]
Ok but this doesn't make sense to me... I mean when you apply a profile with multiple IP addresses which one will be applied to the OS? And secondly, to me it seems completely superfluous since you can just create multiple profiles with different ips

**Comment by KurtisLiggett** [comment]
I think you misunderstand. Here is my terrible explanation.
Sometimes it is necessary to assign 2 or more IP addresses to a single network adapter. This allows you to be on 2 networks at the same time.
For example: you can have an IP of 192.168.1.1 and 10.225.251.1 at the same time without having to change your IP back and forth.

**Comment by giacomocerquone** [comment]
Oh... My huge ignorance was telling me that being on two networks contemporary  wasn't possible! So sorry haha anyway this is a very interesting feature. After the code-cleanup I could help you here

**Comment by ghost** [comment]
Yes pleaaase. That is the reason I downloaded this tool, as I often have to change the different IP addresses one of the network adapters on this machine has. You will have my eternal gratitiude if you allow me do easily add / remove ip addreses in a nice interface with some overview. Also, please allow for the `/` notation of the subnet mask, for example `192.168.0.70/24`.

**Comment by Otiel** [comment]
I would love to see this feature as well.

**Comment by KurtisLiggett** [comment]
I've started working on the backend of this. Just need to make a nice GUI for it.

If I see a lot of interest, I'll put more effort into getting this done sooner.

**Comment by saau** [comment]
I would like love to see this feature as well.

**Comment by KurtisLiggett** [comment]
_From @raym0nde commented on Nov 30, 2017_
_Copied from duplicate issue #58._

I have an application that requires two IP addresses on a single adapter. It would be great if SIC had the ability to set multiple IPs.


![image](https://user-images.githubusercontent.com/24615894/33429990-ddb8b99a-d59c-11e7-9d15-c81059bce09a.png)

**Comment by madbanzai** [comment]
Any update on this feature? It would be great to see it soon.

**Comment by ObiRAM** [comment]
And since you're processing the advanced tab, why not go all the way and support multiple gateways with their own metric. opened an issue for it, but it just might fall into this.

Best regards

**Comment by White-Goblin** [comment]
I'm also requesting the ability to assign multiple IP addresses to a single NIC. I use software which requires this.

**Comment by quarky42** [comment]
I see now that there was already a thread about multiple IPs on a single NIC.  I should have posted here.

My main question:  As a shortcut, could this be implemented without the GUI support at first?   Example:  If we know the proper way to edit the INI file, could we go ahead and take an existing profile and add the second IP and second Subnet to it?   (If we knew the delimeter, say a comma or something, we could just add that delimeter, add the second entry and save the file.)

The GUI would need to be able to not barf on having an unusual entry there, but if the backend code could just read the profile.ini, see that there are two IPs for that interface and assign them to the selected NIC, that would be great.   Ultimately, being able to edit the IPs in the interface would be preferred and having something similar to the Windows interface for that would be great, but the GUI is a nice to have where as the ability to assign two IPs to the NIC would be a great to have feature for me.

Without redesigning the GUI completely, one could just simply add an "Add" / "Del" buttons, some navigation arrows, and a visual indicator of how many IPs are currently being stored for that one profile.     Clicking the Add button would save the current screen, increment the index/counter, provide blank text entry boxes, and the user could then enter the next values.    The Del button could delete the currently displayed values.  The navigation arrows would update the IP and Subnet fields to show the values associated with the currently selected index/counter.    It's not as pretty as a list, but maybe easier to just add a handful of buttons to get this feature out the door?

**Comment by hurr1k4ne** [comment]
![2020-04-21 11_59_08-Window](https://user-images.githubusercontent.com/3234219/79852957-b2388300-83c7-11ea-8b8b-cf02cb4c0338.png)

Same Problem here.
At my work i need the .125. Network to reach the Gateway and Attendees Network.
In addition i need the .120. Network to get access to the business administrations network. 

SECURITY by Complexity....

INI profile edit style like @KurtisLiggett is more than sufficent for me.

**Reply by razorrazor** [reply]
Hi there! 5 years later this feature is not implemented! I've been investigating to help a bit...

From a cmd window, you can add a new IP address (i.e. 192.168.1.10) to an adapter (i.e. named 'Ethernet') like this:
`Netsh int ipv4 add address name="Ethernet" 192.168.1.10 255.255.255.0 SkipAsSource=True`

From a powershell window, you can do the same thing like this:
`New-NetIPAddress -IPAddress 192.168.1.10 -PrefixLength 24 -InterfaceAlias "Ethernet" -SkipAsSource $True`

The flag '_SkipAsSource_' tells you if the address have to be considered as the main output address. The flag '_PrefixLength_' apply the correct subnet mask (i.e. 24 = 255.255.255.0).

So that, Simple IP Config codemasters only have to modify the source code to give the ability of asking for a comma-separated IP list and then to make a call to some of commands above, applying the SkipAsSource=False only to the first address of that list. The config would be applied to the adapter name selected at the top of the SimpleIPConfig interface.

I hope this help and we can get this feature soon.

Regards!

**Comment by KurtisLiggett** [comment]
@razorrazor 
Thanks I appreciate all the investigation and feedback/suggestions.
I think (can't remember now) I had the backend code ready to add multiple IPs, but designing the frontend configuration is a daunting task. Maybe I will look into this again. I may be able to at least get it to the point of allowing someone to manually edit the INI file to add multiple addresses.

I'll keep this post updated here and add any test versions I come up with.

**Reply by razorrazor** [reply]
It would be a temporary workaround to be able to edit the INI file with something like `IpAddressesAlt=<List of comma-separated IP>`.

**Comment by CraigBures** [comment]
Love this app. Just found this thread. This feature would be very helpful. Hope you are able to add it.

**Reply by JonnyTech** [reply]
Is this implemented by editing the .ini file directly?

**Comment by KurtisLiggett** [comment]
I would love to implement this feature, but always get stuck on how to present it (not creative enough!).
If someone would like to sketch up an example frontend of how we could implement this, that would help a lot! I want to keep it "simple" so it would be best to avoid having to do a lot of clicking.

**Reply by quarky42** [reply]
There's already a "standard" established.  Just follow that... If you don't want to do that, then add a couple buttons like I explained without changing the overall look of your UI.

Either way, something 6 years ago that wasn't perfect would have been better than the vaporware that there is now.

I really wanted to use your utility, but couldn't because of this missing feature.  I instead wrote one myself using HTML + VBS.  It wasn't pretty, but it got the job done.

**Reply by KurtisLiggett** [reply]
Thanks for the feedback. Glad to hear you found a solution that works for you.

**Comment by W1BTR** [comment]
This feature is a big one for me. I have to change IPs weekly to new sites. Often there are multiple networks Im connecting to via the same interface. Other than this missing feature, this program is perfect for me (well, I wouldnt mind a "favorites" feature too, but that's not a _need_ like this feature is).

What about having a little "MORE IPs" button to the right of the automatic / static selecter with a little number indicating how many (if any) additional IPs are set, then just a simple list menu with IPs and subnet masks. Additionally, being able to add more gateways in there would be nice. Obviously it'd be greyed out if using DHCP
![image](https://github.com/user-attachments/assets/5eacabd7-92b6-4c69-a338-16b255b47049)

**Reply by W1BTR** [reply]
lol, put it in the wrong spot. Corrected: 
![image](https://github.com/user-attachments/assets/1de9c6de-36e8-4a18-b9b0-e911fab4a95b)

**Comment by vedantraval** [comment]
Here in 2025 with the same requirement as other posters. Is there any expected development for this feature? I regularly keep switching 2+ IPs on multiple network adapters, and having a profile with the secondary IPs saved would be a huge time saver.

Thanks

**Comment by W1BTR** [comment]
Following up on this, definitely a must have for infra work! I have anywhere from 1-6 IPs per adapter at a time, and this changes daily depending on the site.

**Comment by joelkeasley** [comment]
I work with cameras and this is a necessity feature For me as well.   I work with multiple vlans at a single location and required to be on one for internet but need to be able to scan other vlans to find them 
<img width="1270" height="612" alt="image" src="https://github.com/user-attachments/assets/d7b564c5-6e57-4937-a47e-43389f55f5b9" />
